pymc_experimental/model/marginal_model.py

```diff
@@ -244,6 +248,127 @@ def marginalize(self, rvs_to_marginalize: Union[TensorVariable, Sequence[TensorV
         # Raise errors and warnings immediately
         self.clone()._marginalize(user_warnings=True)
 
+    def unmarginalize(
+        self, idata, var_names=None, include_samples=False, extend_inferencedata=True
+    ):
+        """Computes log-likelihoods of marginalized variables conditioned on parameters
```

Comment:
Does this handle multiple marginalized variables?

Reply:
Yes! Though no test yet for that

Reply:
Can you add some info on the docstrings saying how you're handling. Sounds like you are doing one variable at a time evaluation while keeping the remaining ones marginalized?

Reply:
Yes that's exactly what I do. I sample each discrete variable while the rest of them marginalized